files/usr/share/jeos-firstboot/jeos-firstboot-functions

```diff
@@ -26,14 +26,26 @@ list=
 password=''
 modules=()
 
-if pushd "${pfx:-/usr}/share/jeos-firstboot/modules" &>/dev/null; then
-        for module in *; do
-                if [ -f "${module}" ] && source "${module}"; then
-                        modules+=("${module}")
-                fi
-        done
-        popd &>/dev/null
-fi
+_find_modules() {
+	local d f
+	for d in "/etc/jeos-firstboot/modules" "${pfx:-/usr}/share/jeos-firstboot/modules"; do
+		for f in "$d"/*; do
+			if [ -L "$f" ] || [ -f "$f" ]; then
+				echo "${f##*/}" "$f"
+			fi
+		done
+	done
+}
+
+_init_modules() {
+	[ -z "$modules" ] || return 0
+	local module f
+	while read -r module f; do
+		[ -L "$f" ] && [ "$(readlink "$f")" = "/dev/null" ] && continue
```

Comment:
IMO this needs to be a full if. The `&&` shorthand syntax should only be used once per line.

Reply:
:shrug: